src/material/form-field/_form-field-fill-theme.scss

```diff
@@ -97,7 +97,7 @@ $fill-dedupe: 0;
 
       // Server-side rendered matInput with a label attribute but label not shown
       // (used as a pure CSS stand-in for mat-form-field-should-float).
-      .mat-input-server[label]:not(:label-shown) + .mat-form-field-label-wrapper
+      .mat-input-server[label]:not(:placeholder-shown) + .mat-form-field-label-wrapper
```

Comment:
I think we should remove all the `:not(:placeholder-shown)` usages altogether. The recommended approach for labelling inputs has been using `mat-label` for a long time now.

Reply:
Shall we just remove all these rules then? I guess they should not work anyway (see my tiny [copen](https://codepen.io/martin-havala/pen/vYewgBX))

Nevertheless, couldn't we just fix the obvious bug(reverting the pseudo-selector) and handle any further optimizations afterwards?

Reply:
I would rather keep the selectors without the `:not`. It's possible that this was a regression the entire time, but we never noticed it.